File: java/src/main/java/libldt3/model/objekte/BAK.java
/*
 * Copyright 2016-2023  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import java.util.List;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.Kontext;
import libldt3.model.enums.DarstellungErgebniswerte;
import libldt3.model.enums.EinheitMesswert;
import libldt3.model.regel.kontext.K002;
import libldt3.model.regel.kontext.K054;
import libldt3.model.regel.kontext.K100;

/**
 * In diesem Objekt werden die Ergebnisse bakteriologischer Untersuchungen
 * strukturiert abgebildet.
 */
@Objekt(value = "0072", kontextregeln = K100.class)
public class BAK implements Kontext {

    @Objekt
    public static class BAK_DarstellungErgebniswerte implements Kontext {
        @SuppressWarnings("unused")
        public DarstellungErgebniswerte value;
        @Feld(value = "8420", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 60)
        public List<ErgebnisWert> ergebnisWert;
    }

    @Objekt(kontextregeln = {K002.class, K054.class, K100.class})
    public static class ErgebnisWert implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "8419", feldart = Feldart.bedingt_kann)
        @Regelsatz(laenge = 1)
        public EinheitensystemMesswerteWertes einheitensystemMesswerteWertes;
        @Feld(value = "8142", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 10)
        public List<Normalwert> normalwert;
        @Feld(value = "8237", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 12)
        public Fliesstext ergebnistext;
    }

    @Objekt(kontextregeln = K002.class)
    public static class EinheitensystemMesswerteWertes implements Kontext {
        @SuppressWarnings("unused")
        public EinheitMesswert value;
        @Feld(value = "8421", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 60)
        public String masseinheitMesswerteWertes;
    }

    @Feld(value = "8245", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 12)
    public Fliesstext bakErgebnis;
    @Feld(value = "7306", feldart = Feldart.kann)
    @Regelsatz(laenge = 2)
    public List<BAK_DarstellungErgebniswerte> darstellungErgebniswerte;
    @Feld(value = "8246", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 29)
    public Fliesstext bakErgebniswertbezogeneHinweise;

}

File: cs/libldt3/model/objekte/BAK.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.enums;
using libldt3.model.regel.kontext;

namespace libldt3
{
    namespace model
    {
        namespace objekte
        {
            /// <summary>
            /// In diesem Objekt werden die Ergebnisse bakteriologischer Untersuchungen
            /// strukturiert abgebildet.
            /// </summary>
            [Objekt(Value = "0072", Kontextregeln = new[] { typeof(K100) })]
            public class BAK : Kontext
            {
                [Objekt]
                public class BAK_DarstellungErgebniswerte : Kontext
                {
                    public DarstellungErgebniswerte? Value;
                    [Feld(Value = "8420", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public IList<ErgebnisWert> ErgebnisWert;

                }
                [Objekt(Kontextregeln = new[] { typeof(K002), typeof(K054), typeof(K100) })]
                public class ErgebnisWert : Kontext
                {
                    public string Value;
                    [Feld(Value = "8419", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Laenge = 1)]
                    public EinheitensystemMesswerteWertes EinheitensystemMesswerteWertes;
                    [Feld(Value = "8142", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 10)]
                    public IList<Normalwert> Normalwert;
                    [Feld(Value = "8237", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 12)]
                    public Fliesstext Ergebnistext;

                }
                [Objekt(Kontextregeln = new[] { typeof(K002) })]
                public class EinheitensystemMesswerteWertes : Kontext
                {
                    public EinheitMesswert? Value;
                    [Feld(Value = "8421", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string MasseinheitMesswerteWertes;

                }
                [Feld(Value = "8245", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 12)]
                public Fliesstext BakErgebnis;
                [Feld(Value = "7306", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 2)]
                public IList<BAK_DarstellungErgebniswerte> DarstellungErgebniswerte;
                [Feld(Value = "8246", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 29)]
                public Fliesstext BakErgebniswertbezogeneHinweise;

            }
        }
    }
}
